Yona/Yona/MeDashBoard/YonaUserProfileViewController.swift

```diff
@@ -276,43 +276,28 @@ class YonaUserProfileViewController: UIViewController, UITableViewDelegate, UITa
      }
     
     func isUserDataValid() -> validateError {
-        if aUser?.firstName.characters.count == 0 {
-            self.displayAlertMessage("", alertDescription:
-                NSLocalizedString("enter-first-name-validation", comment: ""))
+        if aUser?.firstName.count == 0 {
+            self.displayAlertMessage("", alertDescription:NSLocalizedString("enter-first-name-validation", comment: ""))
             return .firstname
-        }
-        else if aUser?.lastName.characters.count == 0 {
-            self.displayAlertMessage("", alertDescription:
-                NSLocalizedString("enter-last-name-validation", comment: ""))
+        } else if aUser?.lastName.count == 0 {
+            self.displayAlertMessage("", alertDescription: NSLocalizedString("enter-last-name-validation", comment: ""))
             return .lastname
-        } else if aUser?.mobileNumber.characters.count == 0 {
-            self.displayAlertMessage("", alertDescription:
-                NSLocalizedString("enter-number-validation", comment: ""))
+        } else if aUser?.mobileNumber.count == 0 {
+            self.displayAlertMessage("", alertDescription:NSLocalizedString("enter-number-validation", comment: ""))
             return .phone
-        } else if aUser?.nickname.characters.count == 0 {
-            self.displayAlertMessage("", alertDescription:
-                NSLocalizedString("enter-nickname-validation", comment: ""))
+        } else if aUser?.nickname.count == 0 {
+            self.displayAlertMessage("", alertDescription: NSLocalizedString("enter-nickname-validation", comment: ""))
             return .nickname
         } else {
-            var number = ""
-            if let mobilenum = aUser?.mobileNumber {
-                number =  mobilenum
-                
-                let trimmedWhiteSpaceString = number.removeWhitespace()
-                let trimmedString = trimmedWhiteSpaceString.removeBrackets()
-                
-                if trimmedString.validateMobileNumber() == false {
-                    self.displayAlertMessage("", alertDescription:
-                        NSLocalizedString("enter-number-validation", comment: ""))
+            if var mobilenum = aUser?.mobileNumber {
+                mobilenum = mobilenum.removeWhitespace().removeBrackets().formatDutchCountryCodePrefix()
+                if mobilenum.validateMobileNumber() == false {
```

Comment:
Please rename ``validateMobileNumber`` into ``isValidMobileNumber`` and change the logic to simply match this regular expression: ``^\+\d{6,20}$``. This demands a string that starts with a ``+`` and has only digits after that, at least 6 and at most 20. That way, you have a reasonable idea that they've entered a phone number. The server will validate that it's a real mobile number.

Reply:
Updated code.